sdk/plugin/grpc_events.go

```diff
@@ -37,6 +37,10 @@ type GRPCEventsServer struct {
 }
 
 func (s *GRPCEventsServer) SendEvent(ctx context.Context, req *pb.SendEventRequest) (*pb.Empty, error) {
+	if s.impl == nil {
```

Comment:
I think it will also be necessary to do this in `sdk/framework/backend.go`, since when the KV plugin is loaded as an internal plugin, it will use the framework method directly and still return the error from there.

Reply:
Good question. FWIW in my testing, this change did suppress the warning log, so although it's a little weird, the builtin KV plugin is hitting this code path.

Currently we return a similar but slightly different error: https://github.com/hashicorp/vault/blob/670c9522a554b78ad47824de11a8344ff9fe88c6/sdk/framework/backend.go#L696-L698

And the KV plugin ignores that error: https://github.com/hashicorp/vault-plugin-secrets-kv/blob/405c5cb0ec328e4c54682c3c6738120408a35221/backend.go#L455-L462

I like the idea that plugins can tell when the events system is disabled, but that doesn't have to be by getting a sentinel error back from the `Send` method. We could add a separate method for that if we find a concrete use for it. I'm kind of torn whether it's best to be consistent by also changing the returned error to be nil, or just leave it be and let plugins decide what they'd like to do with the error. WDYT?

Reply:
Ah, cool. I think using `ErrNoEvents` as a sentinel is fine to return to signal that the events system is not available.